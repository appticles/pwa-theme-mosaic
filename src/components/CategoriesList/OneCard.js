import React from 'react';
import { Grid, GridRow } from 'semantic-ui-react';

import { categoryPropType } from '../../views/CategoriesCarousel/reducer';
import CategoryCard from '../CategoryCard/index';
import './style.css';

const OneCard = props => (
  <Grid className="full-row">
    <GridRow>
      <CategoryCard category={props.categoriesList[0]} key={Math.random()} />
    </GridRow>
  </Grid>
);

OneCard.propTypes = {
  categoriesList: categoryPropType.isRequired,
};
export default OneCard;
