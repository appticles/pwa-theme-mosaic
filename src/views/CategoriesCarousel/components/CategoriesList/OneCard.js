import React from 'react';
import PropTypes from 'prop-types';
import { Grid, GridRow, GridColumn } from 'semantic-ui-react';

import { categoryPropType } from '../../reducer';
import CategoryCard from '../CategoryCard/index';
import './style.css';

const OneCard = props => (
  <Grid className="full-row">
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[0]} key={props.categoriesList[0].id} />
      </GridColumn>
    </GridRow>
  </Grid>
);

OneCard.propTypes = {
  categoriesList: PropTypes.arrayOf(categoryPropType).isRequired,
};
export default OneCard;
