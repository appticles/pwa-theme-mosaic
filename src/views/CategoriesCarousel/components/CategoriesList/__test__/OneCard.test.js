import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import OneCard from '../OneCard';

describe('One card', () => {
  it('should render correctly', () => {
    const mockCategory = [
      {
        slug: 'mock-slug-1',
        id: 1,
        name: 'Some mock name',
        link: 'http://localhost/mock-slug-1',
      },
    ];
    const output = shallow(<OneCard categoriesList={mockCategory} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
