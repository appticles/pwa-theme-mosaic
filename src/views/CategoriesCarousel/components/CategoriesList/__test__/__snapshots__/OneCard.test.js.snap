// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`One card should render correctly 1`] = `
<Grid
  className="full-row"
>
  <GridRow>
    <GridColumn>
      <CategoryCard
        category={
          Object {
            "id": 1,
            "link": "http://localhost/mock-slug-1",
            "name": "Some mock name",
            "slug": "mock-slug-1",
          }
        }
        key="1"
      />
    </GridColumn>
  </GridRow>
</Grid>
`;
